packages/shared/src/components/squads/SquadJoinBanner.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ReactElement } from 'react';
+import { useMutation } from 'react-query';
+import { useRouter } from 'next/router';
+import { Origin } from '../../lib/analytics';
+import { Button } from '../buttons/Button';
+import { SimpleSquadJoinButton } from './SquadJoinButton';
+import { Squad } from '../../graphql/sources';
+import usePersistentContext from '../../hooks/usePersistentContext';
+import { useJoinSquad } from '../../hooks';
+import { labels } from '../../lib';
+import { useToastNotification } from '../../hooks/useToastNotification';
+import SourceButton from '../cards/SourceButton';
+import useMedia from '../../hooks/useMedia';
+import { mobileL } from '../../styles/media';
+
+export type SquadJoinBannerProps = {
+  squad: Squad;
+  analyticsOrigin: Origin;
+  storageKey: string;
+};
+
+export const SquadJoinBanner = ({
+  squad,
+  analyticsOrigin,
+  storageKey,
+}: SquadJoinBannerProps): ReactElement => {
+  const isMobile = !useMedia([mobileL.replace('@media ', '')], [true], false);
```

Comment:
This is custom size from useSidebarRender?

Reply:
Not quite, because this only needs to applied on mobile due to source bar size issues.

Reply:
Ah got it, that case it's fine